openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -220,16 +236,61 @@ const GlossaryTermTab = ({
 
   const expandableConfig: ExpandableConfig<ModifiedGlossaryTerm> = useMemo(
     () => ({
-      ...getTableExpandableConfig<ModifiedGlossaryTerm>(),
+      ...getTableExpandableConfig<ModifiedGlossaryTerm>(true),
       onExpand: (isOpen, record) => {
         if (isOpen) {
-          fetchGlossaryTerm({ parent: record.id }, true);
+          fetchGlossaryTerm({ parent: record.id }, true, false);
         }
       },
     }),
     [fetchGlossaryTerm]
   );
 
+  const handleMoveRow = useCallback(
+    async (dragRecord: GlossaryTerm, dropRecord: GlossaryTerm) => {
+      if (dragRecord.id === dropRecord.id) {
+        return;
+      }
+
+      setMovedGlossaryTerm({
+        from: dragRecord,
+        to: dropRecord,
+      });
+      setIsModalOpen(true);
+    },
+    []
+  );
+
+  const handleChangeGlossaryTerm = async () => {
+    if (movedGlossaryTerm) {
+      setIsTableLoading(true);
+      const updatedGlossaryTerm = {
+        ...movedGlossaryTerm.from,
+        parent: {
+          fullyQualifiedName: movedGlossaryTerm.to.fullyQualifiedName,
+        },
+      };
+      const jsonPatch = compare(
+        movedGlossaryTerm.from as GlossaryTerm,
```

Comment:
Do we need this type casting?

Reply:
correct, we can avoid let me have a look